examples/manifold/plot_t_sne_perplexity.py

```diff
@@ -0,0 +1,99 @@
+"""
+=============================================================================
+ t-SNE: The effect of various perplexity values on the shape
+=============================================================================
+
+An illustration of t-SNE on the two concentric circles and the S-curve
+datasets for different perplexity values.
+
+We observe the tendency of higher variations for small perplexity values and
+clearer shapes as the value increases.
+
+The size, the distance and the shape of clusters may vary upon initialization,
+perplexity values and does not always convey a meaning.
+
+As shown below, t-SNE for higher perplexities finds meaningful topology of
+two concentric circles, however the size and the distance of the circles varies
+slightly from the original. In contrary to two circles dataset, the shapes
+visually diverge from S-curve topology on S-curve dateset even for larger
+perplexity values.
+"""
```

Comment:
You probably want a mention of the blog post that @amueller gave in the issue, i.e. something like:

For more details, `"How to Use t-SNE Effectively" <http://distill.pub/2016/misread-tsne/>`_
provides a good discussion of the effects of the various parameters, as well
as interactive plots to explore the effects of different parameters.

Not an expert on t-SNE though so I am not able to comment much further on the PR.